src/design-pages/passwordless_gdm.rst

```diff
@@ -0,0 +1,461 @@
+            "role": "smartcard",
+            "selectable": true,
+            "init_prompt": "Insert smartcard",
+            "pin_instruction": "Enter PIN in reader"
+          }
+        },
+        "priority": [
+          "smartcard:1",
+          "smartcard:2"
+        ]
+      }
+    }
+
+GDM replies with the selection:
+
+..  code-block:: json
+
+    {
+      "auth-selection": {
+        "status": "Ok",
+        "smartcard:1": {
+          "pin": "ThePIN"
+        }
+      }
+    }
+
+Passkey
++++++++
+SSSD provides the available authentication mechanisms:
+
+..  code-block:: json
+
+    {
+      "auth-selection": {
+        "mechanisms": {
+          "passkey": {
+            "name": "passkey",
+            "role": "passkey",
+            "selectable": true,
+            "init_instruction": "Insert security key",
+            "pin_request": true,
+            "pin_attempts": 8,
+            "pin_prompt": "Security key PIN",
+            "touch_instruction": "Touch security key"
+          }
+        },
+        "priority": [
+          "passkey"
+        ]
+      }
+    }
+
+GDM replies with the selection:
+
+..  code-block:: json
+
+    {
+      "auth-selection": {
+        "status": "Ok",
+        "passkey": {
+          "pin": "ThePIN"
+        }
+      }
+    }
+
+All authentication mechanisms
++++++++++++++++++++++++++++++
+SSSD provides the available authentication mechanisms:
+
+..  code-block:: json
+
+    {
+      "auth-selection": {
+        "mechanisms": {
+          "password": {
+            "name": "Password",
+            "role": "password",
+            "selectable": true,
+            "prompt": "Password:"
+          },
+          "eidp": {
+            "name": "Web Login",
+            "role": "eidp",
+            "selectable": true,
+            "init_prompt": "Login",
+            "link_prompt": "Login online with another device",
+            "uri": "https://short.url.com/tmp",
+            "code": "1234"
+          },
+          "smartcard:1": {
+            "name": "smartcard ID1",
+            "role": "smartcard",
+            "selectable": true,
+            "init_prompt": "Insert smartcard",
+            "pin_prompt": "PIN"
+          },
+          "passkey": {
+            "name": "passkey",
+            "role": "passkey",
+            "selectable": true,
+            "init_instruction": "Insert security key",
+            "pin_request": true,
+            "pin_attempts": 8,
+            "pin_prompt": "Security key PIN",
+            "touch_instruction": "Touch security key"
+          }
+        },
+        "priority": [
+          "passkey",
+          "eidp",
+          "smartcard:1",
+          "password"
+        ]
+      }
+    }
+
+GDM replies with the selection, as explained in the preceding sections.
+
+Authentication mechanisms priority
+**********************************
+
+The priority for the authentication mechanisms set in the JSON message is a
+hardcoded value and follows the next priority:
```

Comment:
Hi,

why is this not configurable? And how does this relate to the GDM feature to save the last used method. Will the priority only be used if there is no saved method or will the priority always overwrite the last saved method? Or would it make sense to have an option to control this as well?

bye,
Sumit

Reply:
Hi,

I wonder if there might be a use-case where an environment might want to emphasize a specific authentication method but allow others as a fall back. So that always the company's preferred method is shown by default independent what the user used last time. It might be possible to implement this with a `forced` flag or similar which is send in `priority` together with the list of methods? But I'm not sure if this is a valid use-case or not.

bye,
Sumit

Reply:
As I thought this use case is very hypothetical. That's why I think it's better to leave it hardcoded for the moment, and if in the future we see the need to tune it we can always add a new option to change these values.

I have updated this paragraph to mention the GDM feature to remember the authentication mechanism. Please take a look at it.

Reply:
Thanks